driver-async/src/main/com/mongodb/async/client/gridfs/GridFS.java

```diff
@@ -0,0 +1,205 @@
+package com.mongodb.async.client.gridfs;
+
+import com.mongodb.async.SingleResultCallback;
+import com.mongodb.async.client.MongoCollection;
+import com.mongodb.async.client.MongoDatabase;
+import com.mongodb.client.model.CreateIndexOptions;
+import com.mongodb.client.result.DeleteResult;
+import org.bson.Document;
+import org.bson.types.ObjectId;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+
+
+public class GridFS {
+
+    private static final int DEFAULT_CHUNK_SIZE = 255 * 1024;
+    private static final String DEFAULT_BUCKET_NAME = "FS";
+
+    private final int chunkSize;
+    private final MongoDatabase db;
+    private final MongoCollection<GridFSFile> fileCollection;
+    private final MongoCollection<Document> chunkCollection;
+    private final String bucketName;
+
+    public GridFS(final MongoDatabase db, final String bucket, final Integer chunkSize) {
+
+        Objects.requireNonNull(db);
```

Comment:
While it would be nice to be able to use the Objects class, we have decided not to rely on any Java 7 APIs except AsynchronousSocketChannel (which can be replaced by Netty).